app/src/main/java/com/example/cookbook/data/repositories_impl/RecipeRepositoryImpl.kt

```diff
@@ -21,65 +20,58 @@ class RecipeRepositoryImpl @Inject constructor(
     private val database: DatabaseSource
 ) : RecipeRepository {
 
-    override suspend fun refreshDatabaseWithRandomRecipes() {
-        withContext(Dispatchers.IO) {
-            if (database.getAllRecipes().isEmpty()) {
-                val list = service.getRandomRecipeList().hits?.map {
-                    responseToDataMapper(it)
-                } ?: throw Exception(ERROR_MESSAGE)
-                val listOfEntities = list.map {
-                    dataToEntityMapper(it)
-                }
-                database.insertAllRecipes(*listOfEntities.toTypedArray())
+    override fun refreshDatabaseWithRandomRecipes(): Completable {
+        return service.getRandomRecipeList().flatMapCompletable {
+            val list = it.hits?.map { response ->
+                responseToDataMapper(response)
+            } ?: throw Exception(ERROR_MESSAGE)
+            val listOfEntities = list.map { recipe ->
+                dataToEntityMapper(recipe)
             }
+            database.insertAllRecipes(*listOfEntities.toTypedArray())
+            Completable.complete()
         }
     }
 
-    override suspend fun getRecipeList(): List<RecipeData> {
-        return withContext(Dispatchers.IO) {
-            database.getAllRecipes().map { entityToDataMapper(it) }
-        }
-    }
-
-    override suspend fun getRecipeListSync(): LiveData<List<RecipeData>> {
-        return Transformations.map(database.getAllRecipesSync()) {
+    override fun getRecipeListSync(): Observable<List<RecipeData>> {
+        return database.getAllRecipesSync().map {
             it.map { entity ->
                 entityToDataMapper(entity)
             }
         }
     }
 
-    override suspend fun getRecipeListBySearching(q: String): List<RecipeData> {
-        return withContext(Dispatchers.IO) {
-            service.getRecipeListBySearching(q).hits?.map {
-                responseToDataMapper(it)
-            } ?: throw Exception(ERROR_MESSAGE)
+    override fun getRecipeListBySearching(q: String): Single<List<RecipeData>> {
+        return service.getRecipeListBySearching(q).map { listResponse ->
+            listResponse.hits?.map { itemResponse ->
+                responseToDataMapper(itemResponse)
+            }
```

Comment:
Where you subsribe to result of this Single ?

Reply:
In SearchViewModel
fun searchRecipes(q: String) {
        recipeRepository.getRecipeListBySearching(q)
            .subscribeOn(schedulerIo)
...